src/pymor/models/transfer_function.py

```diff
@@ -249,17 +256,27 @@ def bode_plot(self, w, mu=None, ax=None, Hz=False, dB=False, deg=True, adaptive_
         artists
             List of matplotlib artists added.
         """
+        num_input = self._cal_plot_dimension(input_indices, self.dim_input)
+        num_output = self._cal_plot_dimension(output_indices, self.dim_output)
+        if input_indices is None:
+            input_indices = list(range(self.dim_input))
+
+        if output_indices is None:
+            output_indices = list(range(self.dim_output))
```

Comment:
Since you convert the `indices` here anyway, the check for `None` could be removed in `_calc_plot_dimension` if the method is called after the conversion (unless I am missing something). This would make `_calc_plot_dimension` so slim that it could actually be removed.

On the flip side: If the conversion is done in `_calc_plot_dimension` (we would need a different name then) they could be returned and reused, for example in a `sigma_plot` method. I'm thinking something like this:

```python
num_input, input_indices = self._parse_indices(input_indices, self.dim_input)
```

What do you think?

Reply:
Yes. IMO this would be more readable and shorter overall.

Reply:
the only use was preventing redundancies but I think removing it will make the code more readable. I will work on it